src/components/LinkableText.tsx

```diff
@@ -9,27 +9,35 @@ interface Props {
 
 export const LinkableText: React.FC<Props> = ({text, ...props}) => {
   const handlePress = (url: string) => {
-    Linking.openURL(url);
+    Linking.canOpenURL(url).then((supported) => {
+      console.log(supported);
+      const regex = /(http|https)/;
+      if (!regex.test(url)) {
+        const urls = `http://${url}`;
+        Linking.openURL(urls);
+      }
+      Linking.openURL(url);
+    });
   };
 
   // Regular expression to match URLs
-  const urlRegex = /(https?:\/\/[^\s]+)/g;
+  const urlRegex = /^(?:(?:https?):\/\/)?(?:www\.)?[a-zA-Z0-9-]+(?:\.[a-zA-Z]{2,})+(\.\/*[^\s]*)?$/;
 
   // Split the text by URLs
-  const parts = text.split(urlRegex);
+  const parts = text.split(' ');
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The regular expression used for matching URLs seems overly complex and might not cover all possible cases. Also, splitting the text by spaces instead of URLs might lead to incorrect results if the text contains URLs without spaces around them. Consider revising your approach here.
```diff
-  const urlRegex = /^(?:(?:https?):\/\/)?(?:www\.)?[a-zA-Z0-9-]+(?:\.[a-zA-Z]{2,})+(\.\/*[^\s]*)?$/;
-  const parts = text.split(' ');
+  const urlRegex = /(https?:\/\/[^\s]+)/g;
+  const parts = text.split(urlRegex);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->